flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SortMergeResultPartition.java

```diff
@@ -160,18 +160,12 @@ public void setup() throws IOException {
         }
 
         int numRequiredBuffer = bufferPool.getNumberOfRequiredMemorySegments();
-        String errorMessage =
-                String.format(
-                        "Too few sort buffers, please increase %s to a larger value (more than %d).",
-                        NettyShuffleEnvironmentOptions.NETWORK_SORT_SHUFFLE_MIN_BUFFERS,
-                        2 * expectedWriteBuffers);
-        if (numRequiredBuffer < 2 * expectedWriteBuffers) {
-            LOG.warn(errorMessage);
-        }
-
         int numWriteBuffers = Math.min(numRequiredBuffer / 2, expectedWriteBuffers);
         if (numWriteBuffers < 1) {
-            throw new IOException(errorMessage);
+            throw new IOException(
+                    String.format(
+                            "Too few sort buffers, please increase %s.",
+                            NettyShuffleEnvironmentOptions.NETWORK_SORT_SHUFFLE_MIN_BUFFERS));
```

Comment:
I think this it might not be clearer than the before. The old message tells user increase to what size of the network buffer but the new message does not tell the size of network buffer that user should increase to.

Reply:
The error only occurs when only 1 buffer is configured which is rare and increasing it to any value would be OK. Besides, we already give a suggested value in the config option description which is good enough and should never trigger any warnings or exceptions.

Reply:
Thanks for explaining.